### PR TITLE
Issue #3439399: Message module view corrupt after update in some cases

### PR DESCRIPTION
**Pull request by robertragas**
## Problem
In a recent Open Social update the message module went from 1.4 to 1.5. This introduced a change in a view with update hook message_update_8105.

This assumes you have the views.view.message but in our case we noticed that some did not have it, likely due to a corruption in a past Open Social update. This causes the view to be partially created with just the parameters from the update hook. This can then cause a WSOD.

## Solution
We need to re-import the view. Because not everyone might have this issue we also need to check if the view is corrupt before re-importing.

## Issue tracker
https://www.drupal.org/project/social/issues/3439399

## How to test

- [ ] Install Open Social
- [ ]  Remove views.view.message from the config table
- [ ]  Execute message_update_8105 again and notice when you would then load the config it has a corrupt view
- [ ]  Go to this branch, and run drush updb
- [ ] See it does not crash anymore

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] New features or changes to existing features are covered by tests, either unit (preferably) or behat
- [ ] Update path is tested. New hook_updates should respect update order, right naming convention and consider hook_post_update code
- [ ] Module can be safely uninstalled. Update/implement hook_uninstall and make sure that removed configuration or dependencies are removed/uninstalled
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

## Release notes
Internal: We fixed an issue where the message module view would be corrupt as it did not exist and partially imported. This crashes the page with a WSOD. 

PS Which can be solved with a workaround to disable the views_ui

### CODE REVIEW

File: modules/social_features/social_core/social_core.install [hunk 1]
```diff
@@ -314,3 +314,34 @@ function social_core_update_123003(): void {
       ->save();
   }
 }
+
+/**
+ * Recreate views.view.message (module message) if it is corrupted.
+ */
+function social_core_update_123004(): void {
```
Comment (ronaldtebrake) [inline modules/social_features/social_core/social_core.install:321]: We might want to add a dependency just to be sure that it runs after message 8107

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3851-ynnesruudm8rulvgmzt913tdxv4wmhtk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6613f354aa2dcd62df601fb1

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3851-ynnesruudm8rulvgmzt913tdxv4wmhtk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6613f354aa2dcd62df601fb1

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3851-ynnesruudm8rulvgmzt913tdxv4wmhtk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6613f354aa2dcd62df601fb1

File: modules/social_features/social_core/social_core.install [hunk 2]
```diff
@@ -223,6 +223,19 @@ function _social_core_create_menu_links() {
...
 }
 
+/**
+ * Implements hook_update_dependencies().
+ */
+function social_core_update_dependencies(): array {
+  // Update 123003 fixes a view corruption from 8107,
+  // so that one needs to be executed first.
+  $dependencies['social_core'][123003] = [
+    'message' => 8107,
```
Comment (ronaldtebrake) [inline modules/social_features/social_core/social_core.install:233]: https://git.drupalcode.org/project/message/-/blob/8.x-1.x/message.install?ref_type=heads#L52

Isn't it 8105? 🙈

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3851-ynnesruudm8rulvgmzt913tdxv4wmhtk.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6613f354aa2dcd62df601fb1

**Comment by ronaldtebrake** [comment]
cherry-picked to 12.3 and 12.4.x

**Pull request merged**